src/VisualStudio/Core/Next/Remote/ServiceHubRemoteHostClient.WorkspaceHost.cs

```diff
@@ -16,56 +17,49 @@ internal partial class ServiceHubRemoteHostClient
         private class WorkspaceHost : ForegroundThreadAffinitizedObject, IVisualStudioWorkspaceHost, IVisualStudioWorkingFolder
         {
             private readonly VisualStudioWorkspaceImpl _workspace;
-            private readonly RemoteHostClient _client;
+
+            /// <summary>
+            /// The current session we have open to the remote host.  Only accessible from the
+            /// UI thread.
+            /// </summary>
+            private readonly KeepAliveSession _session;
```

Comment:
Note: @heejaechang I need help with the lifetime here.  Will KeepAliveSession automatically clean up after itself when teh WorkspaceHost goes away?